end-user-application-machines/new-agent-vm/Install-Idle-Shutdown.sh

```diff
@@ -0,0 +1,282 @@
+    fi
+
+    if [ $poll_interval -gt $MAX_POLL_INTERVAL ]; then
+        poll_interval=$MAX_POLL_INTERVAL
+        logit "Reset polling interval to $poll_interval minutes"
+    elif [ $poll_interval -lt $MIN_POLL_INTERVAL ]; then
+        poll_interval=$MIN_POLL_INTERVAL
+        logit "Reset polling interval to $poll_interval minutes"
+    fi
+
+    if [ $cpu_threshold -gt 100 ]; then
+        cpu_threshold=100
+        logit "Reset cpu threshold to $cpu_threshold"
+    elif [ $cpu_threshold -lt 0 ]; then
+        cpu_threshold=0
+        logit "Reset cpu threshold to $cpu_threshold"
+    fi
+}
+
+####################################################################
+# Main program
+# set the globals wait_minutes and poll_interval
+####################################################################
+function main() {
+    wait_minutes=60  # default is an hour
+    poll_interval=15 # default is 15 minutes
+    cpu_threshold=20 # default 20%
+
+    # this will potentially reset the above two variables
+    processArgs $@
+    
+    required_idle_time=$(( 60 * $wait_minutes))
+
+    if [ $(activeConnectionsExist) -gt 0 ]; then
+        echo exit 0
+    fi
+
+    logit "Monitoring system for idle state every $poll_interval minutes and will shutdown machine if idle (every cpu < $cpu_threshold% utilization) for $wait_minutes minutes."
+
+    startTime=0
+    # Main while loop.  Flip between two states: someone is connected, and a triggered timer
+    # counting until we shutdown or someone connects
+    while [ 1 -ne 0 ]; do
+        if [ $(activeConnectionsExist) -gt 0 ]; then
+            startTime=0
+        else
+            if [ $startTime -eq 0 ]; then
+                startTime=$(/bin/date +%s)
+            fi
+
+            cpuLoad=$(getCPULoad)
+            if [ $cpuLoad -ge $cpu_threshold ]; then
+                # a cpu load greater than threshold, assume something's going on
+                #logit "Shutdown process reset because CPU load ($cpuLoad%) > $cpu_threshold%"
+                startTime=0
+            else
+                currentTime=$(/bin/date +%s)
+                elapsedIdleTime=$(( $currentTime - $startTime ))
+
+                if [ $required_idle_time -le $elapsedIdleTime ]; then
+                logit "Shutting down idle machine."
+                sudo /sbin/shutdown now
+                fi
+            fi
+        fi
+
+        /bin/sleep $((60 * $poll_interval))
+    done 
+
+    exit 0
+}
+
+main $@
+EOF
+}
+
+function install() {
+    echo "Installing"
+    touch ${SERVICE_PATH}
+    chmod 664 ${SERVICE_PATH}
+    mkdir -p ${SERVICE_CONFIG_PATH}
+    touch ${SERVICE_CONFIG}
+    chmod 664 ${SERVICE_CONFIG}
+    create_monitor_script
+    cat <<EOF> ${SERVICE_PATH}
+[Unit]
+Description=Teradici CAM Idle Shutdown monitoring service
+After=pcoip.service
+
+[Service]
+Type=simple
+ExecStart=/usr/bin/sudo /usr/bin/bash ${MONITOR_SCRIPT} \${MinutesIdleBeforeShutdown} \${PollingIntervalMinutes} \${CPUUtilizationLimit}
```

Comment:
The service config doesn't specify a user to run this service as, which means the service will run under the logged-in user; since it requires sudo, that means a) the logged-in user requires not just sudo permission but passwordless sudo.

I think it would be better to add a `User` directive to the service config and have this run as root.